platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/conflicts/DefaultCapabilitiesConflictHandler.java

```diff
@@ -42,36 +40,37 @@
 
 public class DefaultCapabilitiesConflictHandler implements CapabilitiesConflictHandler {
     private final List<Resolver> resolvers;
+
+    /**
+     * Tracks capabilities present on nodes as a non-default capability.
+     */
```

Comment:
Also to remove after other work when default works the same as any capability.

Reply:
We'll see if we get that far. I have a feeling the performance optimizations were added there for a reason.